src/core.js

```diff
@@ -1593,8 +1614,16 @@ function _buildSquel(flavour = null) {
       }));
     }
 
-    having (condition, ...values) {
-      this._condition(condition, ...values);
+    having (field, operator, ...values) {
+      let validOperators = ['=', '<', '>', '<=', '>=', '<>', '!=', 'in', 'not in', 'like', 'not like', 'is', 'is not'];
```

Comment:
Looks like we're repeating the code which checks for a valid operator and then building a string out of it. I think this checking code should be shifted into the `BaseBuilder` class. Furthermore, the then-block code should be inside `AbstractConditionBlock` so that all subclasses automatically benefit from it. I think `having()` can then go back to just being a single-line function. Do you agree?